power: Add OS Power Management debug hooks

ramakrishnapallala opened the pull request:
Add the necessary PM hooks for logging and debugging
purposes.

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

pizi-nordic commented on subsys/power/power.c:
```diff
@@ -18,6 +19,78 @@ LOG_MODULE_REGISTER(power);
+{
+
+#ifdef CONFIG_RTC
+	pm_timer = device_get_binding(CONFIG_PM_DEBUG_RTC_DEV_NAME);
```
Do we need dedicated RTC for that? Why we cannot use ticks or system timer cycles?

nashif replied:
agree. Not sure why we are using RTC here, not many platforms have this. Seems to be legacy from when power management was focused on devices with RTC driver (!= system timer)

pizi-nordic commented on subsys/power/power.c:
```diff
@@ -18,6 +19,78 @@ LOG_MODULE_REGISTER(power);
+
+struct pm_debug_info {
+	u32_t count;
+	u32_t last_res;
```
IMHO average will give more information than last.

pizi-nordic commented on subsys/power/power.c:
```diff
@@ -100,6 +180,9 @@ static int sys_pm_init(struct device *dev)
 	ARG_UNUSED(dev);
 
 	sys_pm_create_device_list();
+
```
Empty line not needed.

pizi-nordic reviewed (commented):
Could we register a CLI command to fetch statistics of the power management?

nashif commented:
no need for subsys: in the commit message.

Commit:
power: Add OS Power Management debug hooks

Add the necessary PM hooks like Low Power state entry
count and residencies for logging and debugging purposes.

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

ramakrishnapallala commented:
@pizi-nordic @nashif  I have addressed the comments and updated the PR. Please have a look.
